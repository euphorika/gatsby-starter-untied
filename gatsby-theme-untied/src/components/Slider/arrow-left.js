import React from 'react'
import PropTypes from 'prop-types'

const ArrowLeft = ({ color }) => (
  <span
    dangerouslySetInnerHTML={{
      __html: `<svg id="Ebene_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 84.1 85.5"><style>.st3{clip-path:url(#SVGID_2_);fill:none;stroke:#fff}</style><defs><path id="SVGID_1_" d="M24.1 28.4h35v35h-35z"/></defs><clipPath id="SVGID_2_"><use xlink:href="#SVGID_1_" overflow="visible"/></clipPath><g clip-path="url(#SVGID_2_)"><defs><path id="SVGID_3_" d="M24.1 28.4h35v35h-35z"/></defs><clipPath id="SVGID_4_"><use xlink:href="#SVGID_3_" overflow="visible"/></clipPath><g opacity=".39" clip-path="url(#SVGID_4_)"><defs><path id="SVGID_5_" d="M15.1 27.4h53v43h-53z"/></defs><clipPath id="SVGID_6_"><use xlink:href="#SVGID_5_" overflow="visible"/></clipPath><path clip-path="url(#SVGID_6_)" fill="#fff" d="M24.1 28.4h35v35h-35z"/></g></g><path class="st3" d="M47.9 59.1L35.3 46.5M47.9 34.5L35.3 47.1"/></svg>`,
    }}
  />
)

ArrowLeft.propTypes = {
  color: PropTypes.string,
}

ArrowLeft.defaultProps = {
  color: '#f1f1f1',
}

export default ArrowLeft
