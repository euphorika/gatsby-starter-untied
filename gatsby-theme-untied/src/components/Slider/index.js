/** @jsx jsx */
import React from 'react'
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'

import Slide from './one-slide'
import ArrowLeft from './arrow-left.js'
import ArrowRight from './arrow-right.js'

class Slider extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      images: this.props.images || [],
      width: 100,
      slideIndex: 0,
    }
  }

  slideLeft() {
    this.setState({
      slideIndex:
        this.state.slideIndex === 0
          ? this.state.images.length - 1
          : (this.state.slideIndex - 1) % this.state.images.length,
    })
  }

  slideRight() {
    this.setState({
      slideIndex:
        this.state.slideIndex === this.state.images.length - 1
          ? 0
          : (this.state.slideIndex + 1) % this.state.images.length,
    })
  }

  renderNavigation() {
    return (
      <div
        sx={{
          position: 'absolute',
          top: 'calc(50% - 10px)',
          left: '0',
          right: '0',
          zIndex: '3',
        }}
      >
        <div
          role="button"
          sx={{
            cursor: 'pointer',
            width: '20px',
          }}
          onClick={() => this.slideLeft()}
        >
          <ArrowLeft />
        </div>
        <div
          role="button"
          sx={{
            cursor: 'pointer',
            width: '20px',
          }}
          onClick={() => this.slideRight()}
        >
          <ArrowRight />
        </div>
      </div>
    )
  }

  renderSlides() {
    return (
      <div
        sx={{
          transition: 'margin-left 0.5s',
        }}
        style={{
          width: `${this.state.images.length * this.state.width}%`,
          marginLeft: `-${this.state.width * this.state.slideIndex}%`,
          display: `flex`,
        }}
      >
        {this.state.images.map((image, index) => {
          return <Slide image={image.childImageSharp.fluid} key={index} />
        })}
      </div>
    )
  }

  render() {
    return (
      <div
        sx={{
          width: '100%',
          overflow: 'hidden',
          position: 'relative',
        }}
      >
        <div
          sx={{
            marginBottom: '0',
          }}
        >
          {this.renderSlides()}
          {this.renderNavigation()}
        </div>
      </div>
    )
  }
}

Slider.propTypes = {
  images: PropTypes.array,
}

export default Slider
