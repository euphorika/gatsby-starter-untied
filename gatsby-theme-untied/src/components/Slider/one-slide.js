/** @jsx jsx */
import PropTypes from "prop-types"
import Img from "gatsby-image"

import { jsx } from "theme-ui"

const OneSlide = ({ image }) => (
  <div
    sx={{
      float: "left",
      width: "100%",
    }}
  >
    <Img fluid={image} />
  </div>
)

OneSlide.propTypes = {
  image: PropTypes.object.isRequired,
}

export default OneSlide
