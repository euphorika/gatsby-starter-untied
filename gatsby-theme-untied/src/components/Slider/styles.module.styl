@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.slider
  width 100%
  overflow hidden
  position relative
  margin-bottom 50px

  .slider-arrows
    position absolute
    top calc(50% - 10px)
    left 0
    right 0
    z-index 3

    & > [role="button"]
      cursor pointer
      opacity 0.8
      position absolute

      &.arrow-right
        right 0

@media only screen and (max-width: 500px) 
    & > [role="button"]
      width 20px

  .inner-container
    margin-bottom 0px

    .slider-items
      transition margin-left 0.5s

      .one-slider
        float left
        width 100%