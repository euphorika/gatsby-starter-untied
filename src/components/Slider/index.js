import React from 'react'
import Slide from './one-slide'

import ArrowLeft from './images/arrow-left.png'
import ArrowRight from './images/arrow-right.png'

import styles from './styles.module.styl'

class Slider extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      width: 300,
      height: 300,
      slideIndex: 0,
      images: [
        'nature-1.jpg',
        'nature-2.jpg',
        'nature-3.jpg',
        'nature-4.jpg',
        'nature-5.jpg',
        'nature-6.jpg',
      ],
    }
  }

  slideLeft() {
    this.setState({
      slideIndex:
        this.state.slideIndex === 0
          ? this.state.images.length - 1
          : this.state.slideIndex - 1,
    })
  }

  slideRight() {
    this.setState({
      slideIndex:
        this.state.slideIndex === this.state.images.length - 1
          ? 0
          : this.state.slideIndex + 1,
    })
  }

  renderNavigation() {
    return (
      <div className={styles.sliderArrows}>
        <div
          role="button"
          className={styles.arrowLeft}
          onClick={() => this.slideLeft()}
        >
          <img src={ArrowLeft} alt="" />
        </div>
        <div
          role="button"
          className={styles.arrowRight}
          onClick={() => this.slideRight()}
        >
          <img src={ArrowRight} alt="" />
        </div>
      </div>
    )
  }

  renderSlides() {
    const images = this.state.images
    const slideStyles = {
      marginLeft: `-${this.state.width * this.state.slideIndex}px`,
    }
    return (
      <div className={styles.sliderItems} style={slideStyles}>
        {images.map((image, index) => {
          return (
            <Slide
              image={image}
              width={this.state.width}
              height={this.state.height}
              key={index}
            />
          )
        })}
      </div>
    )
  }

  render() {
    return (
      <div className={styles.slider}>
        {this.renderNavigation()}
        <div
          className={styles.innerContainer}
          style={{
            width: `${this.state.images.length * this.state.width}px`,
          }}
        >
          {this.renderSlides()}
        </div>
      </div>
    )
  }
}

export default Slider
