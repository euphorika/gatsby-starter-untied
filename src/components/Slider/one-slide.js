import React from 'react'
import PropTypes from 'prop-types'

const OneSlide = ({ image, width, height }) => {
  const backgroundImage = require(`./images/${image}`)
  const styles = {
    backgroundImage: `url(${backgroundImage})`,
    backgroundPosition: 'center top',
    backgroundRepeat: 'no-repeat',
    float: 'left',
    width: `${width}px`,
    height: `${height}px`,
  }
  return <div className="slide" style={styles} />
}

OneSlide.propTypes = {
  image: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
}

export default OneSlide
