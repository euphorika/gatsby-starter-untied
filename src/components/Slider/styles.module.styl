@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.slider
  width 100%
  overflow hidden
  position relative

  .slider-arrows
    position absolute
    top 40%
    left 0
    right 0
    z-index 3

    & > [role="button"]
      cursor pointer
      opacity 0.8
      position absolute

      &.arrow-right
        right 0

  .inner-container
    margin-bottom 30px

    .slider-items
      transition margin-left 0.5s

      .one-slider
        float left
        width 100%