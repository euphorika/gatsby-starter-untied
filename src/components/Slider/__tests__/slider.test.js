import React from 'React'
import renderer from 'react-test-renderer'
import Slider from '../'

describe('Slider', () => {
  it('renders correctly', () => {
    const image = {}
    const slider = <Slider title="Title" image={image} />
    const tree = renderer.create(slider).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
