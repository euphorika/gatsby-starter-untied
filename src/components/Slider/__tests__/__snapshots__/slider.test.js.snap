// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<div
  className="slider"
>
  <div
    className="sliderArrows"
  >
    <div
      className="arrowLeft"
      onClick={[Function]}
      role="button"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </div>
    <div
      className="arrowRight"
      onClick={[Function]}
      role="button"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="innerContainer"
    style={
      Object {
        "width": "1800px",
      }
    }
  >
    <div
      className="sliderItems"
      style={
        Object {
          "marginLeft": "-0px",
        }
      }
    >
      <div
        className="slide"
        style={
          Object {
            "backgroundImage": "url(test-file-stub)",
            "backgroundPosition": "center top",
            "backgroundRepeat": "no-repeat",
            "float": "left",
            "height": "300px",
            "width": "300px",
          }
        }
      />
      <div
        className="slide"
        style={
          Object {
            "backgroundImage": "url(test-file-stub)",
            "backgroundPosition": "center top",
            "backgroundRepeat": "no-repeat",
            "float": "left",
            "height": "300px",
            "width": "300px",
          }
        }
      />
      <div
        className="slide"
        style={
          Object {
            "backgroundImage": "url(test-file-stub)",
            "backgroundPosition": "center top",
            "backgroundRepeat": "no-repeat",
            "float": "left",
            "height": "300px",
            "width": "300px",
          }
        }
      />
      <div
        className="slide"
        style={
          Object {
            "backgroundImage": "url(test-file-stub)",
            "backgroundPosition": "center top",
            "backgroundRepeat": "no-repeat",
            "float": "left",
            "height": "300px",
            "width": "300px",
          }
        }
      />
      <div
        className="slide"
        style={
          Object {
            "backgroundImage": "url(test-file-stub)",
            "backgroundPosition": "center top",
            "backgroundRepeat": "no-repeat",
            "float": "left",
            "height": "300px",
            "width": "300px",
          }
        }
      />
      <div
        className="slide"
        style={
          Object {
            "backgroundImage": "url(test-file-stub)",
            "backgroundPosition": "center top",
            "backgroundRepeat": "no-repeat",
            "float": "left",
            "height": "300px",
            "width": "300px",
          }
        }
      />
    </div>
  </div>
</div>
`;
